test/Window/Context.test.cpp

```diff
@@ -1,8 +1,59 @@
 #include <SFML/Window/Context.hpp>
+#include <SFML/Window/ContextSettings.hpp>
 
+#include <doctest/doctest.h>
+
+#include <WindowUtil.hpp>
+#include <string>
 #include <type_traits>
 
 static_assert(!std::is_copy_constructible_v<sf::Context>);
 static_assert(!std::is_copy_assignable_v<sf::Context>);
 static_assert(!std::is_nothrow_move_constructible_v<sf::Context>);
 static_assert(!std::is_nothrow_move_assignable_v<sf::Context>);
+
+TEST_CASE("[Window] sf::Context" * doctest::skip(skipDisplayTests))
+{
+    SUBCASE("Construction")
+    {
+        const sf::Context context;
+
+        CHECK(context.getSettings().majorVersion > 0);
+    }
+
+    SUBCASE("Version String")
+    {
+        sf::Context context;
+
+        CHECK(context.setActive(true));
+
+#if defined(SFML_SYSTEM_WINDOWS)
+        using glGetStringFuncType = const char*(__stdcall*)(unsigned int);
+#else
+        using glGetStringFuncType = const char* (*)(unsigned int);
+#endif
+
+        auto glGetStringFunc = reinterpret_cast<glGetStringFuncType>(sf::Context::getFunction("glGetString"));
+
+        REQUIRE_UNARY(!!glGetStringFunc);
```

Comment:
What does `REQUIRE_UNARY` do?

Reply:
`REQUIRE` aborts the subcase if the check fails, and `_UNARY` omits doctest trying to convert the expression into something it can pretty print. We only care if the pointer is non-null so `!!` converts it into a `bool`.

Reply:
I'm not sure how this improves on `REQUIRE(glGetStringFunc)`. We already do this testing the non-nullness of pointers. The printed output looks like this:

```
 /Users/thrasher/Projects/sfml/test/Network/Packet.test.cpp:98: SUCCESS: CHECK( packet.getData() != nullptr ) is correct!
  values: CHECK( 0x1410043b0 != nullptr )
```

The `_UNARY` suffix gives us this instead

```
/Users/thrasher/Projects/sfml/test/Network/Packet.test.cpp:98: SUCCESS: CHECK_UNARY( packet.getData() != nullptr ) is correct!
  values: CHECK_UNARY( true )
```

which I don't think is any better since you no longer see the code that failed inside the macro. When it fails you get much less information:

```
/Users/thrasher/Projects/sfml/test/Network/Packet.test.cpp:98: ERROR: CHECK_UNARY( packet.getData() == nullptr ) is NOT correct!
  values: CHECK_UNARY( false )
```

Reply:
Those functions return pointers to objects, we have a function pointer here. doctest tries to convert it to `void*` in order to print it as part of the expression decomposition and that will always fail because converting between function pointers and object pointers is something the compiler really tries to prevent you from doing.